design/ceph/ceph-cluster-cleanup.md

```diff
@@ -0,0 +1,65 @@
+# Ceph cluster clean up policy
+
+## Use case
+
+As a rook user, I want to clean up the datadirhostpath after I intentionally uninstall ceph cluster , so that I can start a new cluster without having to do any manual clean up.
+
+## Background
+
+If the user deletes a cluster and starts a new cluster on the same hosts, the path used by dataDirHostPath must be deleted. Otherwise, stale keys and other config will remain from the previous cluster and the new mons will fail to start. As of now, the user has to manually delete the dataDirHostPath on each host. This implementation aims to automate this deletion.
+
+## Design Flow
+
+### Preconditions
+
+- Operator receives a delete request when the user deletes the ceph cluster.
+- **Important**: User confirmation is required before cleaning up the dataDirHostPath. This is important because user might have accidentally deleted the CR and in that case cleaning up the hostpath won’t recover the cluster. This final user confirmation can be a flag/label on the cephCluster
+  Adding the confirmation labels will cause the operator to refuse running an orchestration.
+- Cluster clean up should start when DeletionTimeStamp is observed on the cephCluster CR
+- Cleanup should start once the Finalizers are executed (removed).
+
+### Cleanup Process
+
+- Identify the nodes where ceph daemon are running. This should be done before the finalizers are executed.
+- Create a batch job that runs on each of the above nodes.
```

Comment:
How about listing these items in the order they are executed? It's confusing if we mention the batch job here, but then the next bullet says they wait until the daemons are deployed. The operator is the one waiting before launching the jobs.